inspirehep/modules/disambiguation/records.py

```diff
@@ -0,0 +1,162 @@
+        with additional values for an experiment and a field_categories
+        taken from the hep paper.
+
+        Example:
+            profile = {u'affiliations': [{u'value': u'Yerevan Phys. Inst.'}],
+                       u'alternative_name': None,
+                       u'curated_relation': False,
+                       u'email': None,
+                       u'full_name': u'Chatrchyan, Serguei',
+                       u'inspire_id': None,
+                       u'orcid': None,
+                       u'profile': u'',
+                       u'recid': None,
+                       u'role': None,
+                       u'uuid': u'd63537a8-1df4-4436-b5ed-224da5b5028c'}
+
+    :return:
+        A recid, where the new profile can be accessed.
+
+        Example:
+            "1234"
+    """
+    # Get the author's name.
+    name = profile.get('full_name')
+
+    # Template of an initial record.
+    record = {'collections': [{'primary': 'HEPNAMES'}],
+              'name': {'value': name},
+              '$schema': _get_author_schema()}
+
+    # The author's email address.
+    # Unfortunately the method will not correlate a given e-mail address
+    # with an affiliation.
+    if 'email' in profile:
+        email = profile.get('email')
+
+        record['positions'] = []
+        record['positions'].append({'email': email})
+
+    # The author can be a member of more than one affiliation.
+    if 'affiliations' in profile:
+        affiliations = profile.get('affiliations')
+
+        if 'positions' not in record:
+            record['positions'] = []
+
+        for affiliation in affiliations:
+            name = affiliation.get('value')
+            recid = affiliation.get('recid', None)
+
+            if recid:
+                record['positions'].append(
+                    {'institution': {'name': name, 'recid': recid}})
+            else:
+                record['positions'].append(
+                    {'institution': {'name': name}})
+
+    # FIXME: The method should also collect the useful data
+    #        from the publication, like category field, subject,
+    #        etc.
+
+    # Disconnect the signal on insert of a new record.
+    after_record_insert.disconnect(append_new_record_to_queue)
+
+    # Create a new author profile.
+    record = Record.create(record, id_=None)
+
+    # Create Inspire recid.
+    record_pid = inspire_recid_minter(record.id, record)
+
+    # Extend the new record with Inspire recid and self key.
+    record['control_number'] = record_pid.pid_value
+    record['self'] = inspire_dojson_utils.get_record_ref(
+        record_pid.pid_value, 'authors')
+
+    # Apply the changes.
+    record.commit()
+    db.session.commit()
+
+    # Add the record to Elasticsearch.
+    indexer = RecordIndexer()
+    indexer.index_by_id(record_pid.object_uuid)
+
+    # Reconnect the disconnected signal.
+    after_record_insert.connect(append_new_record_to_queue)
```

Comment:
All this logic that deals with low level implementation details (pids, receivers, ES...) of the creation of an author record should be abstracted in an helper (if it wasn't already).